src/main/java/org/cactoos/text/TextEnvelope.java

```diff
@@ -73,23 +75,36 @@ public final int hashCode() {
         return new UncheckedScalar<>(this.origin).value().hashCode();
     }
 
-    // @todo #942:30min Refactor TextEnvelope.equals(). Current implementation
-    //  of TextEnvelope.equals() uses some things that we must avoid, like more
-    //  than one return on method, instance of usage and typecasting. Refactor
-    //  this method so we can get rid of these smelly things.
-    //
     @Override
-    @SuppressWarnings("PMD.OnlyOneReturn")
     public final boolean equals(final Object obj) {
-        if (this == obj) {
-            return true;
-        }
-        if (!(obj instanceof Text)) {
-            return false;
-        }
-        final Text that = (Text) obj;
-        return new UncheckedText(this).asString().equals(
-            new UncheckedText(that).asString()
+        return new UncheckedScalar<>(
+            new Or(
+                () -> this == obj,
+                this.equalsOther(obj)
```

Comment:
@strokyl any reason why you split this over three methods?

Reply:
@strokyl please leave it all inside the `equals` method

* I found no explicit mention of this by Yegor, but I believe he also frowns upon compound names for methods (so do I). If you look at the code produced in his other projects you'll see how this idea is implicitly enforced throughout.
* Personally I find these extra methods unnecessary - the implementation required here isn't too complex to justify them

Reply:
Ok I will rework that tonight. Thank you for your feedback

Reply:
done